derive/src/to_delegate.rs

```diff
@@ -169,9 +171,7 @@ impl RpcMethod {
 			special_args.iter().find(|(_,sty)| sty == ty).is_none());
 
 		let tuple_fields : &Vec<_> =
-			&(0..param_types.len() as u8)
-				.map(|x| ident(&((x + 'a' as u8) as char).to_string()))
-				.collect();
+			&(TUPLE_FIELD_NAMES[0..param_types.len()].iter().map(|name| ident(name)).collect());
```

Comment:
it will panic if `param_types.len() > TUPLE_FIELD_NAMES.len()` (maybe return an error "max supported params number is 26"?), also can be simplified to
```suggestion
			&(TUPLE_FIELD_NAMES.iter().take(param_types.len()).map(|name| ident(name)).collect());
```

Reply:
@ordian my guess is that before it was trying to use `{` as an type param identifier instead of plain panicking, which would also fail, but with more cryptic message.

Reply:
@kirushik yes, but note that `take` won't panic (it will take `min(TUPLE_FIELD_NAMES.len(), param_types.len())`, also note that serde::deserialize is not implemented for tuples of length > 16.

[Here](https://github.com/paritytech/jsonrpc/compare/kirushik/would_you_like_some_assistance_today...ao-error_on_too_many_params) is a patch to return an error, feel free to use it.